src/Uno.UI/UI/Xaml/Controls/Image/Image.iOS.cs

```diff
@@ -312,35 +312,39 @@ private void UpdateLayerRect(Size availableSize)
 				return;
 			}
 
-			var parameters = (availableSize, Image.Size, HorizontalAlignment, VerticalAlignment, Stretch);
-			if (parameters == _layerLastParameters)
-			{
-				return; // nothing to update
-			}
-
-			_layerLastParameters = parameters;
-
 			var imageSize = Image.Size.ToFoundationSize();
 
-			var sourceRect = new Rect(Point.Zero, imageSize);
+			// Calculate the resulting space required on screen for the image
+			var containerSize = this.MeasureSource(availableSize, imageSize);
+
+			// Calculate the position of the image to follow stretch and alignment requirements
+			var contentRect = this.ArrangeSource(availableSize, containerSize);
 
-			this.MeasureSource(availableSize, ref sourceRect);
-			this.ArrangeSource(availableSize, ref sourceRect);
+			// Calculate the required container to position the image in the AvailableSize
+			var containerRect = new Rect(default, availableSize);
+			containerRect.Intersect(contentRect);
 
-			var relativeX = sourceRect.X / imageSize.Width;
-			var relativeY = sourceRect.Y / imageSize.Height;
+			var relativeX = (contentRect.X - containerRect.X) / contentRect.Width;
```

Comment:
Could `contentRect.Width` or `contentRect.Width` be zero ?

Reply:
Yes. That was the last bug I was seeking. It's could happen when using `UniformToFill`.

Reply:
Oooh you means the width.  Nope they can't. If you're here, you have an image with a size of at least one pixel.